src/transformers/trainer.py

```diff
@@ -1968,8 +1976,12 @@ def _inner_training_loop(
                 dist.barrier()
             elif is_sagemaker_mp_enabled():
                 smp.barrier()
-
             self._load_best_model()
+        elif args.load_best_model_at_end and self.state.best_model_checkpoint is None:
+            logger.warning(
+                'Best model cannot be loaded because "trainer.state.best_model_checkpoint" is None. '
+                'Have you set the "save_strategy" Training Argument?'
```

Comment:
For my own understanding - what values of `"save_strategy"` are expected to work or not work here? If we're to emit a log here - we should be explicit about exactly what the user has to do. Ideally we'd check that save strategy is compatible with `load_best_model_at_end`.

Reply:
Values of `save_strategy` that would work are `save_strategy="epoch"` OR, if `save_strategy="steps"`, then `save_steps` should be set lower than `max_steps`.

However, since adding the check in `_inner_training_loop`, the problematic behavior shouldn't be occurring, so it doesn't seem necessary to specify. Perhaps removing the line that reads, 'Have you set the "save_strategy" Training Argument?' Or we can leave out the log warning entirely, if you think that's best.